File: backend/app/schemas/education.py
from typing import List
from sqlmodel import SQLModel

class QuizQuestionRead(SQLModel):
    id: int
    prompt: str
    options: List[str]

class ModuleRead(SQLModel):
    id: int
    title: str
    content: str
    questions: List[QuizQuestionRead]
    badge_earned: bool = False

class QuizSubmission(SQLModel):
    answers: List[int]

class QuizResult(SQLModel):
    score: int
    passed: bool
    badge_awarded: bool

class BadgeRead(SQLModel):
    id: int
    name: str
    module_id: int | None = None

class ModuleUpdate(SQLModel):
    enabled: bool

File: frontend/src/api/education.ts
import type { ApiClient } from './client'

export interface QuizQuestion {
  id: number
  prompt: string
  options: string[]
}

export interface EducationModule {
  id: number
  title: string
  content: string
  questions: QuizQuestion[]
  badge_earned: boolean
}

export interface QuizResult {
  score: number
  passed: boolean
  badge_awarded?: boolean
}

export interface Badge {
  id: number
  name: string
  module_id?: number | null
}

export const listEducationModules = (client: ApiClient) =>
  client.get<EducationModule[]>('/education/modules')

export const submitModuleQuiz = (client: ApiClient, moduleId: number, answers: number[]) =>
  client.post<QuizResult>(`/education/modules/${moduleId}/quiz`, { answers })

export const listMyBadges = (client: ApiClient) =>
  client.get<Badge[]>('/education/badges/me')
